packages/smart-accounts/src/mav-2/utils.ts

```diff
@@ -0,0 +1,512 @@
+import type { UpgradeToData } from "@aa-sdk/core";
+import {
+  concatHex,
+  toHex,
+  type Address,
+  type Chain,
+  type Hex,
+  encodeFunctionData,
+  parseAbi,
+  hexToNumber,
+  size,
+  type Hash,
+  encodeAbiParameters,
+  stringToHex,
+} from "viem";
+import type { GetAccountParameter } from "../types";
+import type { SmartAccount } from "viem/account-abstraction";
+import { semiModularAccountStorageAbi } from "./abis/semiModularAccountStorageAbi.js";
+import {
+  HookType,
+  SignaturePrefix,
+  type HookConfig,
+  type ModuleEntity,
+  type ValidationConfig,
+} from "./types.js";
+import type { ModularAccountV2Base } from "./accounts/base";
+import * as WebAuthnP256 from "ox/WebAuthnP256";
+import * as Signature from "ox/Signature";
+
+export const DEFAULT_OWNER_ENTITY_ID = 0;
+
+export const EXECUTE_USER_OP_SELECTOR: Hex = "0x8DD7712F";
+
+export type PackUOSignatureParams = {
+  // orderedHookData: HookData[];
+  validationSignature: Hex;
+};
+
+// Signature packing utility for user operations
+export const packUOSignature = ({
+  // orderedHookData, TODO: integrate in next iteration of MAv2 sdk
+  validationSignature,
+}: PackUOSignatureParams): Hex => {
+  return concatHex(["0xFF", "0x00", validationSignature]);
+};
+
+// TODO: direct call validation 1271
+export type Pack1271SignatureParams = {
+  validationSignature: Hex;
+  entityId: number;
+};
+
+// Signature packing utility for 1271 signatures
+export const pack1271Signature = ({
+  validationSignature,
+  entityId,
+}: Pack1271SignatureParams): Hex => {
+  return concatHex([
+    "0x00",
+    toHex(entityId, { size: 4 }),
+    "0xFF",
+    SignaturePrefix.EOA,
+    validationSignature,
+  ]);
+};
+
+export const getDefaultMAV2FactoryAddress = (chain: Chain): Address => {
+  // TODO(jh): do we really need this if it's the same on all chains? should we do similar to LightAccount?
+  switch (chain.id) {
+    default:
+      return "0x00000000000017c61b5bEe81050EC8eFc9c6fecd";
+  }
+};
+
+export const getDefaultSMAV2BytecodeAddress = (chain: Chain): Address => {
+  switch (chain.id) {
+    default:
+      return "0x000000000000c5A9089039570Dd36455b5C07383";
+  }
+};
+
+export const getDefaultSMAV2StorageAddress = (chain: Chain): Address => {
+  switch (chain.id) {
+    default:
+      return "0x0000000000006E2f9d80CaEc0Da6500f005EB25A";
+  }
+};
+
+export const getDefaultSMAV27702Address = (chain: Chain): Address => {
+  switch (chain.id) {
+    default:
+      return "0x69007702764179f14F51cdce752f4f775d74E139";
+  }
+};
+
+export const getDefaultMAV2Address = (chain: Chain): Address => {
+  switch (chain.id) {
+    default:
+      return "0x00000000000002377B26b1EdA7b0BC371C60DD4f";
+  }
+};
```

Comment:
the functions that create the `SmartAccount` should just have these internally as defaults, w/ options to manually override them.